plugins/GSdx/GSWndEGL.cpp

```diff
@@ -235,15 +235,19 @@ GSVector4i GSWndEGL::GetClientRect()
 	return GSVector4i(0, 0, w, h);
 }
 
-void GSWndEGL::SetSwapInterval(int vsync)
+void GSWndEGL::SetSwapInterval()
 {
 	// 0 -> disable vsync
 	// n -> wait n frame
-	eglSwapInterval(m_eglDisplay, vsync);
+	eglSwapInterval(m_eglDisplay, m_vsync);
 }
 
 void GSWndEGL::Flip()
 {
+	if (m_vsync_change_requested) {
```

Comment:
What do you think about using an atomic swap ? I think, it shows better the intended behavior.
`if (m_vsync_change_requested.exchange(false))`